node/web/domain/cart/Cart.js

```diff
@@ -0,0 +1,45 @@
+
+module.exports.name = 'Cart';
+module.exports.dependencies = ['Blueprint', 'ObjectID', 'exceptions'];
+module.exports.factory = function (Blueprint, ObjectID, exceptions) {
+    'use strict';
+
+    var Cart = function (cart) {
+
+        var self = {
+            addToCart: undefined,
+            deleteFromCart: undefined
+        };
+        self.totalAmount = cart.totalAmount;
```

Comment:
You aren't defending your signature here. `cart.totalAmount` and `cart.books` could be anything, including `null`. If `self.books` ends up being `null`, then `self.addToCart` with throw and an unhandled exception when you check the length of `self.books` in the for loop.

In testing, I see that each time I add a book to my cart, the books in my cart are overwritten, instead of being appended.